vizro-core/src/vizro/models/_navigation/accordion.py

```diff
@@ -71,29 +71,30 @@ def build(self, *, active_page_id=None):
                     persistence=True,
                     persistence_type="session",
                     always_open=True,
+                    # `active_item` is required to open the accordion automatically when navigating from a homepage
+                    # to any of the pages in the accordion.
                     active_item=active_item,
                 )
             ],
             id="nav-panel",
         )
 
-    def _create_accordion_buttons(self, pages, active_page_id):
-        """Creates a button for each provided page that is registered."""
-        accordion_buttons = []
+    def _create_nav_links(self, pages):
+        """Creates a `NavLink` for each provided page that is registered."""
+        nav_links = []
         for page_id in pages:
             try:
                 page = dash.page_registry[page_id]
             except KeyError as exc:
                 raise KeyError(
                     f"Page with ID {page_id} cannot be found. Please add the page to `Dashboard.pages`"
                 ) from exc
-            accordion_buttons.append(
-                dbc.Button(
+            nav_links.append(
+                dbc.NavLink(
                     children=[page["name"]],
-                    key=page["relative_path"],
-                    className="accordion-item-button",
-                    active=page_id == active_page_id,
+                    className="accordion-item-link",
+                    active="partial",
```

Comment:
This is actually a really nice feature of the `NavLink`, as we don't need the `active_page_id` for this anymore, but it will automatically detect it from the URL. Great push @antonymilne ! 🚀

Reply:
There is one buggy behaviour though - while I think "partial" makes sense in the long-run especially if we might add URLs in the style of `../first-page/sub-page/container` in the future, I think switching to "exact" makes more sense for us now. With "partial" currently the homepage or the first page with href` /` is always active 😅  

Which doesn't work so nicely visually